### DISCUSSION TITLE
Redeploying shared services

### DISCUSSION THREAD

**Discussion by abianche** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#67] [Q&A] [answered]
Hello,

I've made modifications to `auth_manager.py` and have several other updates to the business services. While I can successfully redeploy the business services using code-commit, the current setup for shared_services doesn't support this.

From what I gather, I might need to set up a second pipeline specifically for deploying the modified shared services. Before embarking on that potentially time-consuming task (even if I would need this as well at some point, so if someone has it, that would be great), I'm wondering if there's an alternative, such as tweaking the deployment.sh script? Or any other ways, like download the zip, repackage and re-uploading?

Any guidance would be greatly appreciated. Thank you!

**Comment by ujwalbukka** [comment, marked as answer]
Hi @abianche - You are correct, you need to build a code pipeline for the shared services. Below are alternative approaches.

1. All shared services are part of the [bootstrap-template.yaml](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/server/bootstrap-template.yaml), you just need to redeploy it. You can run command at [line54](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/deployment.sh#L54) and [line55](https://github.com/aws-samples/aws-saas-factory-ref-solution-serverless-saas/blob/main/deployment.sh#L55) from deployment.sh
2. You can just re-run whole deployment.sh too.
